File: data_structures_by_c++/linked_list/singly_linked_list/count_nodes.cpp
#include <bits/stdc++.h>
using namespace std;

// #define int long long

class Node
{
public:
    int val;
    Node *next;
    Node(int val)
    {
        this->val = val;
        this->next = NULL;
    }
};

int node_number(Node *head)
{
    Node *tmp = head;
    int cnt = 0;
    while (tmp != NULL)
    {
        tmp = tmp->next;
        cnt++;
    }
    return cnt;
}

int main()
{
    // ios::sync_with_stdio(false);
    // cin.tie(nullptr);

    Node *head = new Node(10);
    Node *a = new Node(20);
    Node *b = new Node(30);

    head->next = a;
    a->next = b;

    cout << node_number(head);
}

File: data_structures_by_python/linked_list/singly_linked_list/count_nodes.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None


def count_node(head):
    tmp = head
    cnt = 0
    while tmp != None:
        cnt += 1
        tmp = tmp.next
    return cnt


head = Node(10)
a = Node(20)
b = Node(30)

head.next = a
a.next = b

print(count_node(head))